src/hooks/useWalletSupportsChain/useWalletSupportsChain.ts

```diff
@@ -29,33 +29,48 @@ import {
   supportsPolygon,
   supportsThorchain,
 } from '@shapeshiftoss/hdwallet-core'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import { MetaMaskShapeShiftMultiChainHDWallet } from '@shapeshiftoss/hdwallet-shapeshift-multichain'
 import { useMemo } from 'react'
 import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { selectAccountIdsByChainId } from 'state/slices/portfolioSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 type WalletSupportsChainArgs = {
   isSnapInstalled: boolean | null
+  chainAccountIds: AccountId[] // allows dynamic chain-support detection for Ledger
   chainId: ChainId
   wallet: HDWallet | null
 }
 
 // use outside react
 export const walletSupportsChain = ({
   chainId,
+  chainAccountIds,
   wallet,
   isSnapInstalled,
 }: WalletSupportsChainArgs): boolean | null => {
   if (!wallet) return false
   const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
-  // Naming is slightly weird there, but the intent is if this evaluates to false, it acts as a short circuit
-  const shortCircuitFeatureDetection =
-    !isMetaMaskMultichainWallet || (isMetaMaskMultichainWallet && isSnapInstalled)
+  // A wallet may have feature-capabilities for a chain, but not have runtime support for it
+  // e.g MM without snaps installed, or Ledger without chain account ids (meaning the user didn't connect said chain's accounts)
+  const hasRuntimeSupport = (() => {
+    if (Boolean(isLedger(wallet) && !chainAccountIds.length)) return false
+    if (isMetaMaskMultichainWallet && !isSnapInstalled) return false
+    if (!isMetaMaskMultichainWallet) return true
```

Comment:
metamask guranteed to have evm, so  the only runtime check would be for non evm chain ids.
```suggestion
    if (isMetaMask(wallet) && !isSnapInstalled && !isEvmChainId(chainId)) return false
```

Reply:
d403e5c0535812abcd35cb85d1887160e48f597e